xunit.runner.wpf/Impl/RemoteTestUtil.Connection.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Threading;
+using xunit.runner.data;
+using xunit.runner.wpf.ViewModel;
+
+namespace xunit.runner.wpf.Impl
+{
+    internal sealed partial class RemoteTestUtil : ITestUtil
+    {
+        private sealed class Connection : IDisposable
+        {
+            private NamedPipeClientStream _stream;
+            private Process _process;
+            private ClientReader _reader;
+
+            internal NamedPipeClientStream Stream => _stream;
+
+            internal ClientReader Reader => _reader;
+
+            internal Connection(NamedPipeClientStream stream, Process process)
+            {
+                _stream = stream;
+                _process = process;
+                _reader = new ClientReader(stream);
+            }
+
+            internal void Close()
+            {
+                if (_process != null)
+                {
+                    Debug.Assert(_stream != null);
+
+                    try
+                    {
+                        _stream.WriteAsync(new byte[] { 0 }, 0, 1);
```

Comment:
I picked named pipes because it's a very easy method of IPC. The core of the communication though is via `Stream`.  Only the initial connect code deals with named pipes.  We can easily swap this to Stdin/Stoud if we like later and it won't have a large impact on the code.  

As for the methods you suggested: 

- stdin / stdout: It's really hand to use stdout as a diagnostic mechanism for the worker.  For example today if you uncomment the hidden window aspect of the process you can watch the worker run and it produces meaningful debug data.  
- http: No real objection to this.  I think we could use JSON independent of the IPC mechanism.